command/agent/acl_endpoint.go

```diff
@@ -180,6 +185,31 @@ func (s *HTTPServer) ACLTokenSpecificRequest(resp http.ResponseWriter, req *http
 	return s.aclTokenCrud(resp, req, accessor)
 }
 
+func (s *HTTPServer) aclSelfPolicy(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	if req.Method != http.MethodGet {
+		return nil, CodedError(http.StatusMethodNotAllowed, ErrInvalidMethod)
+	}
+
+	policyArgs := structs.GenericRequest{}
+	if s.parse(resp, req, &policyArgs.Region, &policyArgs.QueryOptions) {
+		return nil, nil
+	}
+
+	// Resolve policies for workload identities
+	policyReply := structs.ACLPolicySetResponse{}
+	if err := s.agent.RPC("ACL.GetClaimPolicies", &policyArgs, &policyReply); err != nil {
+		return nil, err
+	}
```

Comment:
My objection to this API as it stands is that we're only presenting the ACL policies for Workload Identity claims but the API is `/v1/acl/policy/self`. Ignoring unifying the CLI output for the moment, either (a) this API should retrieve policies for _any_ authenticated request or (b) it should be under a different route than `/v1/acl/policy/self`. I would prefer (a) because it's much less code to maintain with the only cost being a few extra lines in the RPC handler to make it work.

Reply:
[4053a48](https://github.com/hashicorp/nomad/pull/25588/commits/4053a48a6c18977511d9ddecdd6f8ec7621f4382) implements option (a), but then we end up with a bit of a mess, because `GetClaimPolicies` and `ListPolicies` RPCs return different types (maps or slices, respectively). I thought perhaps the key in the map is the WI name, but it's just policy name, so I'll follow-up with a change that just returns a list of `ACLPolicyStub` to get a unified output.

Reply:
ok @tgross, as discussed today, [da46b31](https://github.com/hashicorp/nomad/pull/25588/commits/da46b310f2c2bca5aaa98a5d81cd32d60817b4c4) unifies the output and adds `JobACL` field to the stub. I think this is an acceptable compromise.